tree-sitter-stack-graphs/src/cli/query.rs

```diff
@@ -73,107 +76,130 @@ pub struct Definition {
+        let mut logger = self.logger.file(&log_path);
+
+        let mut file_reader = FileReader::new();
+        let tag = file_reader.get(&reference.path).ok().map(sha1);
+        match self
+            .db
+            .status_for_file(&reference.path.to_string_lossy(), tag.as_ref())?
+        {
+            FileStatus::Indexed => {}
+            _ => {
+                logger.failure("not indexed", None);
+                return Ok(Vec::default());
             }
+        }
 
-            let lines = PositionedSubstring::lines_iter(source);
-            let mut span_calculator = SpanCalculator::new(source);
+        logger.processing();
 
-            db.load_graph_for_file(&reference.path.to_string_lossy())?;
-            let (graph, _, _) = db.get();
+        self.db
+            .load_graph_for_file(&reference.path.to_string_lossy())?;
+        let (graph, _, _) = self.db.get();
 
-            let reference = match reference.to_assertion_source(graph, lines, &mut span_calculator)
-            {
-                Ok(result) => result,
-                Err(_) => {
-                    logger.failure("invalid file or position", None);
-                    return Ok(());
-                }
-            };
-
-            if reference.iter_references(graph).next().is_none() {
-                logger.failure("no references", None);
-                return Ok(());
-            }
-            let starting_nodes = reference.iter_references(graph).collect::<Vec<_>>();
-
-            let mut actual_paths = Vec::new();
-            let mut reference_paths = Vec::new();
-            match db.find_all_complete_partial_paths(
-                starting_nodes,
-                &stack_graphs::NoCancellation,
-                |_g, _ps, p| {
-                    reference_paths.push(p.clone());
-                },
-            ) {
-                Ok(_) => {}
-                Err(StorageError::Cancelled(..)) => {
-                    logger.failure("path finding timed out", None);
-                    return Ok(());
-                }
-                err => err?,
-            };
-
-            let (graph, partials, _) = db.get();
-            for reference_path in &reference_paths {
-                if reference_paths
-                    .iter()
-                    .all(|other| !other.shadows(partials, reference_path))
-                {
-                    actual_paths.push(reference_path.clone());
-                }
-            }
+        let starting_nodes = reference.iter_references(graph).collect::<Vec<_>>();
+        if starting_nodes.is_empty() {
+            logger.warning("no references", None);
+            return Ok(Vec::default());
+        }
 
-            if actual_paths.is_empty() {
-                logger.warning("no definitions", None);
-                return Ok(());
+        let mut actual_paths = Vec::new();
+        let mut reference_paths = Vec::new();
+        match self.db.find_all_complete_partial_paths(
```

Comment:
This is the method that assumes that you've already loaded everything you need into the `Database` instance, right?  Not the phased algorithm?  But if I'm tracking, it also looks like you're only loading in the stack subgraph for the file containing the reference, which makes me think this isn't going to produce any cross-file results.

Reply:
Not in this case, because this is the `SQLiteReader::find_all_complete_partial_paths` method. This one uses `ForwardPartialPathStitcher` internally do load more data in between phases.